### PR TITLE
drivers: serial: uart_max32: Fixing bugs and code optimizations

### PR DESCRIPTION
**Pull request by hfakkiz**
Defined 'api_configure' function outside of 'CONFIG_UART_USE_RUNTIME_CONFIGURE' macro because it is the only configuration function and used in device initialization function.
Removed unnecessary 'IF_ENABLED' part.

### CODE REVIEW

File: drivers/serial/uart_max32.c [hunk 1]
```diff
@@ -443,9 +441,9 @@ static const struct uart_driver_api uart_max32_driver_api = {
 		.uart_conf.flow_ctrl =                                                             \
 			DT_INST_PROP_OR(_num, hw_flow_control, UART_CFG_FLOW_CTRL_NONE),           \
 		IF_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN,                                           \
-			   (.irq_config_func = uart_max32_irq_init_##_num,))};                     \
+			   (.irq_config_func = uart_max32_irq_init_##_num,))};                    \
 	static struct max32_uart_data max32_uart_data##_num = {                                    \
-		IF_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN, (.cb = NULL,))};                          \
+		IF_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN, (.cb = NULL,))};                         \
```
Comment (MaureenHelm) [inline drivers/serial/uart_max32.c:446]: Revert this change

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_max32.c (hunk 1 above)
Comment (hfakkiz) [reply drivers/serial/uart_max32.c]: Thank you, I reverted it.

**Review by MaureenHelm — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]
